common/scala/src/main/scala/whisk/core/entity/Exec.scala

```diff
@@ -93,9 +94,18 @@ protected[core] case class JavaExec(code: Attachment[String], main: String) exte
 
 protected[core] case class PythonExec(code: String) extends CodeExec[String](Exec.PYTHON)
 
-protected[core] case class BlackBoxExec(image: String) extends Exec(Exec.BLACKBOX) {
-    def size = image sizeInBytes
-    override val sentinelledLogs = false
+/**
+ * @param image the image name
+ * @param code an optional script or zip archive (as base64 encoded) string
+ */
+protected[core] case class BlackBoxExec(override val image: String, code: Option[String]) extends CodeExec[Option[String]](Exec.BLACKBOX) {
```

Comment:
it is tested - i also think it is desirable.
it is not documented (neither is the intended feature yet beyond this pull request comments - I can add the bits to the actions docs.)